### PR TITLE
add tensorboard and add export sample for mel and audio 

### PR DESCRIPTION
**Pull request by lpscr**
@SWivid hi i add sample export and tensorboard 

when you call finetune-cli with `--export_samples` you exprot audio and mel 
when you call finetune-cli with `--logger` choices=["none", "wandb", "tensorboard"]
by defualt its `wandb` dont broken stuff 

image

here also new gradio finetune gui ;)

image

please check my code in trainer if all ok 

thank you

### CODE REVIEW

**Comment by ZhikangNiu** [comment]
@SWivid @lpscr https://huggingface.co/docs/accelerate/usage_guides/tracking

maybe the following code is more simple
```python
accelerator = Accelerator(log_with="all") 
```

**Comment by lpscr** [comment]
@SWivid  can you make this update like you want ? 
because you know beter i dont know a lot about llm stuff , accelerator , etc

**Comment by SWivid** [comment]
@lpscr In general, we don't put a lot of burden on the training process, preferring to do things like visualizations only when needed. 
Specifically, if we do not have a network, we will choose the offline wandb, and then copy it out and make it online.

We felt we should keep the trainer itself simple and include only the necessary functionality. For example, able to switch to tensorboard to record the loss, and other visualizations could be included in specific scripts.

**Comment by lpscr** [comment]
@SWivid ok check if all ok now

**Comment by lpscr** [comment]
@SWivid i just add option also for logger

image

i finish all update

**Comment by SWivid** [comment]
I will modify based on your last update.
Some questions: `normalize_and_colorize_spectrogram` is for what usage?

**Comment by lpscr** [comment]
@SWivid basic the only need 

i want to give color in mel i dont if there is better way i dont normalized i get like this

image

**Comment by SWivid** [comment]
@lpscr if are to cut, we should just remove this melspec, leave only audio. otherwise the melspec is useless.

**Comment by lpscr** [comment]
@SWivid you know beter , do like you think best , the mel i want to see to compare how clear it's the gen with original and  how close ,
 
you mean you remove you can see in tensorboard after ? or export

**Comment by SWivid** [comment]
> the mel i want to see to compare how clear it's the gen with original and how close

it is more straightforward to listen
will not see big similarity between mels, or say if very simillar is not a good thing

**Comment by lpscr** [comment]
here one fix i working on to fix glitsh effect in begin the mel help me to see this 

fix_begin_error_problem

fix_begin_error

**Comment by SWivid** [comment]
it's not a big deal, we are just to see if the model can generate something intelligible.

**Comment by lpscr** [comment]
@SWivid basic very easy fix this you need clear pause begin end , 
add little pause 50msec in end for the ref audio , like this the ref have pause in end and dont make glitsh

**Comment by SWivid** [comment]
@lpscr I mainly do pruning of code
1. samples saving process kept
2. remove logging to tensorboard files, cuz it's a duplicate (already saved once)
3. take off melspec as not very helpful to determine whether training goes well

**Comment by lpscr** [comment]
@SWivid great thank you goind test now and i tell you

**Comment by lpscr** [comment]
@SWivid great look like all working great i just test it !  thank you very much ! now the finetune gradio best all automatic :)
i want to make new clear tutorials tomorrow

**Comment by SWivid** [comment]
@lpscr Yes, you could directly update the readme in `train/` dir

**Comment by lpscr** [comment]
@SWivid check this video tutorial i hope you like enjoy ; ) 
https://github.com/SWivid/F5-TTS/discussions/143
